llama_index/vector_stores/pinecone.py

```diff
@@ -210,21 +213,30 @@ def __init__(
     ) -> None:
         """Initialize params."""
         try:
-            import pinecone
+            # import pinecone
+            pinecone = __import__("pinecone")
         except ImportError:
             raise ImportError(import_err_msg)
 
         if pinecone_index is not None:
-            self._pinecone_index = cast(pinecone.Index, pinecone_index)
+            self._pinecone_index = cast(pinecone.Index, pinecone_index)  # type: ignore[name-defined]
         else:
-            if index_name is None or environment is None:
-                raise ValueError(
-                    "Must specify index_name and environment "
-                    "if not directly passing in client."
-                )
+            if hasattr(pinecone, "version") and pinecone.version >= "3.0.0":
+                if index_name is None:
```

Comment:
NIT / question unrelated to your PR or changes: 

When I look at this code it makes me feel like the index_name should probably be validated elsewhere - because we're duplicating similar code and expanding the total number of lines of code (and impacting legibility imho). 

Do you feel the same way? Might be worth noting as an issue to improve in a future PR, if you feel the same way.

Reply:
I do agree! I'll open an issue for future dev work